src/infinistore.cpp

```diff
@@ -155,12 +150,24 @@ Client::~Client() {
         tcp_recv_buffer_ = NULL;
     }
 
+    while (!outstanding_rdma_ops_queue_.empty()) {
+        auto item = outstanding_rdma_ops_queue_.front();
+        delete[] item.first;   // ibv_send_wr array
+        delete[] item.second;  // ibv_sge array
+        outstanding_rdma_ops_queue_.pop_front();
+    }
     destroy_rdma_context(&rdma_ctx_);
 }
 
 void on_close(uv_handle_t *handle) {
     client_t *client = (client_t *)handle->data;
-    delete client;
+
+    if (client) {
+        client->handle_ = NULL;
```

Comment:
what's the point of assigning a member variable right before `delete` ing its enclosing instance?

Reply:
i see. thanks. just a nit, might want to use `nullptr` in lieu of `NULL` in modern C++ code.